cmd/nerdctl/run_restart.go

```diff
@@ -0,0 +1,65 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/runtime/restart"
+)
+
+func generateRestartOpts(ctx context.Context, client *containerd.Client, restartFlag, logURI string) ([]containerd.NewContainerOpts, error) {
+	policySlice := strings.Split(restartFlag, ":")
+	switch policySlice[0] {
+	case "", "no":
+		return nil, nil
+	case "on-failure", "unless-stopped":
+		version, err := client.Version(ctx)
+		if err != nil {
+			return nil, err
+		}
+		ver, err := semver.NewVersion(version.Version)
+		if err != nil {
+			return nil, err
+		}
+		ver163 := semver.MustParse("v1.6.3")
+		if ver.LessThan(ver163) {
+			return nil, fmt.Errorf("containerd version %s does not support restart policy %s", version.Version, policySlice[0])
+		}
```

Comment:
For the main branch, it seems to always be v1.6.0-xxx.
```
$ containerd --version
containerd github.com/containerd/containerd v1.6.0-290-g0c2b6a102 0c2b6a102a3a40819c28234ac41eec6c919d6b2f
```

Reply:
Please check the plugin metadata, not the daemon version

https://github.com/containerd/containerd/blob/7f8506a4f96d5de14a94d47b28265ac1652a61c1/runtime/restart/monitor/monitor.go#L76

Reply:
Makes sense